assignments/lambda-classes.js

```diff
@@ -1 +1,110 @@
-// CODE here for your Lambda Classes
+class Person {
+  constructor(attributes) {
+    this.name = attributes.name;
+    this.location = attributes.location;
+    this.age = attributes.age;
+    this.gender = attributes.gender;
+  }
+  speak() {
+    return `Hello my name is ${this.name} and I am from ${this.location}.`;
+  }
+}
+
+class Instructor extends Person {
+  constructor(childAttributes) {
+    super(childAttributes);
+    this.favLanguage = childAttributes.favLanguage;
+    this.specialty = childAttributes.specialty;
+    this.catchPhrase = childAttributes.catchPhrase;
+  }
+  demo(subject) {
+    return `Today we are learning about ${subject}`;
+  }
+  grade(student, subject) {
+    return `${student} receives a perfect score on ${subject}`;
+  }
+}
+
+class Student extends Instructor {
+  constructor(grandChildAttributes) {
+    super(grandChildAttributes);
+    this.previousbackground = grandChildAttributes.previousbackground;
+    this.className = grandChildAttributes.className;
+    this.favSubjects = grandChildAttributes.favSubjects;
+  }
+  listsSubject() {
+    return `These are my favorite subjects in Lambda School: ${this.favSubjects}`;
+  }
+  PRAssignment() {
+    return `${this.name} has submitted a PR for ${this.subject}`;
+  }
+  sprintChallenge() {
+    return `${this.name} has begun spring challenge ${this.subject}`;
+  }
+}
+
+class ProjectManager extends Student {
+  constructor(greatGrandChildAttributes) {
+    super(greatGrandChildAttributes);
+    this.gradClassName = greatGrandChildAttributes.gradClassName;
+    this.favInstructor = greatGrandChildAttributes.favInstructor;
+  }
+  standUp(channel) {
+    return `${this.name} announces to ${this.channel}, @channel standy times!​​​​​`;
+  }
+  debugsCode() {
+    return `${this.name} debugs ${andrew.name} 's code on ${this.subject}`;
```

Comment:
what is andrew.name doing here? Debug code is supposed to accept a student and a subject.